### DISCUSSION TITLE
Please support alternative date formats

### DISCUSSION THREAD

**Discussion by Lewis8265** [sysadminsmedia/homebox#318] [Ideas]
Please add support for the DD/MM/YYYY date format :)

**Comment by PAS-BC** [comment]
Agreed.
The Purchase and Warranty dates are very confusing when you enter the date correctly (as dd/mm/yyyy) but they are displayed as mm/dd/yyyy.

'Group Settings' allow for a localised Currency - could this be expanded to localised date format please @tankerkiller125 ?

**Reply by tankerkiller125** [reply]
While I implemented the initial localization for languages in the frontend, the currencies were before my time, and now most frontend changes are handled by @tonyaellie I generally avoid the frontend when I can because frankly I suck at it. It should in theory be possible (given the backend just spits out ISO formatted dates), but I personally won't make any promises on that front, I have no idea how complicated that may or may not be.

**Reply by PAS-BC** [reply]
Thank you for the quick reply and explanation.
We'll see whether @tonyaellie has a 'quick fix' solution (fingers crossed).

**Reply by tonyaellie** [reply]
As we use date-fns already for some things we could reuse it and either allow the user to input like DD/MM/YYYY in whatever order they want or/and we could have a list of presets for the user to choose from.

**Reply by PAS-BC** [reply]
@tonyaellie sounds like a solution is within your grasp ?
A list of presets would be pretty short list I expect hehe.

**Reply by tonyaellie** [reply]
Ok so having now investigated this we do actually kind of support this already (https://github.com/sysadminsmedia/homebox/blob/main/frontend/composables/use-formatters.ts#L30-L83) the issue is that if you set your lang to say English it will always use en-US, basically we need a way to set the locale more specifically, maybe have an option to override the locale?

**Reply by PAS-BC** [reply]
You often see options in other Apps, like:
English (Australia)
English (UK)
English (US)

For English (Australia) set date to 'dd/mm/yyyy' 

Done :)

**Reply by tonyaellie** [reply]
https://github.com/sysadminsmedia/homebox/pull/648

**Comment by PAS-BC** [comment]
Many thanks for the great date improvement.
